File: app/src/main/java/com/wb/logistics/ui/flights/delegates/FlightsAdapter.java
package com.wb.logistics.ui.flights.delegates;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.wb.logistics.R;
import com.wb.logistics.databinding.FlightsLayoutRouteLegendBinding;

import java.util.List;

public class FlightsAdapter extends ArrayAdapter<String> {

    @NonNull
    private final List<String> items;
    @NonNull
    private final OnItemClickCallBack onItemClickCallBack;

    public interface OnItemClickCallBack {

        void onItemClick(int idItem);

    }

    public FlightsAdapter(@NonNull Context context, @NonNull List<String> items,
                          @NonNull OnItemClickCallBack onItemClickCallBack) {
        super(context, R.layout.flights_layout_route_legend);
        this.items = items;
        this.onItemClickCallBack = onItemClickCallBack;
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        ViewHolder holder;
        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(R.layout.flights_layout_route_legend, null);
            holder = new ViewHolder(convertView);
            convertView.setTag(holder);
        } else {
            holder = (ViewHolder) convertView.getTag();
        }
        String item = items.get(position);
        holder.binding.routeLegendDescriptionText.setText(item);
        return convertView;
    }

    @Override
    public int getCount() {
        return items.size();
    }

    private class ViewHolder {

        FlightsLayoutRouteLegendBinding binding;

        protected ViewHolder(View rootView) {
            binding = FlightsLayoutRouteLegendBinding.bind(rootView);
        }

    }

}

File: app/src/main/java/ru/wb/perevozka/ui/flights/delegates/FlightsAdapter.kt
package ru.wb.perevozka.ui.flights.delegates

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import ru.wb.perevozka.R
import ru.wb.perevozka.databinding.FlightsRouteItemLayoutBinding

class FlightsAdapter(context: Context, private val items: List<String>) :
    ArrayAdapter<String>(context, R.layout.flights_route_item_layout) {

    override fun isEnabled(position: Int): Boolean {
        return false
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val view: View
        val holder: ViewHolder
        if (convertView == null) {
            view =
                LayoutInflater.from(context).inflate(R.layout.flights_route_item_layout, null)
            holder = ViewHolder(view)
            view.tag = holder
        } else {
            view = convertView
            holder = view.tag as ViewHolder
        }
        holder.binding.routeLegendDescriptionText.text = items[position]
        return view
    }

    override fun getCount(): Int {
        return items.size
    }

    private inner class ViewHolder(rootView: View?) {
        var binding: FlightsRouteItemLayoutBinding = FlightsRouteItemLayoutBinding.bind(rootView!!)
    }

}